### PR TITLE
Port documentation to CMake [10864]

### PR DESCRIPTION
**Pull request by EduPonz**
This PR ports the documentation over to CMake:
* Makefile no longer exists, so its commands are not valid anymore
* Build is performed with either `cmake --build` or `colcon build`. This PR adds build options:
    * `BUILD_DOCUMENTATION` -> Whether to build the sphinx docs. Defaults ON
    * `COMPILE_TESTS` -> Whether to build the tests. Defaults ON
* With `colcon`/`cmake`, the documentation is built using Sphinx's HTML builder. Users can still use other builders using Sphinx directly.
* Tests are moved to CTest so they can be run with `ctest` or `colon test`
* Documentation is installed
* README.md has been updated to reflect the new building methods

Merging this PR entails that:
* Current CI jobs will need to be updated.
* It offers the possibility to test the documentation with every Fast DDS PR, since now it can be built and test with `colcon`. Doing so will prevent merging Fast DDS PR with spelling errors in the doxygen comments, which right now breaks the documentation tests every now and then.
* External contributions are made easier, since with this PR, everyone can build and run the documentation tests locally, whereas before there was a dependency with eProsima's intranet.

### CODE REVIEW

File: code/grep.py
Comment (JLBuenoLopez-eProsima) [inline code/grep.py:1]:
```suggestion
# Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: code/CMakeLists.txt
Comment (JLBuenoLopez-eProsima) [inline code/CMakeLists.txt:142]:
```suggestion
# Check that all cpp snippets are on the CPP files and not in the RST files so they are compiled
```

File: code/CMakeLists.txt
Comment (JLBuenoLopez-eProsima) [inline code/CMakeLists.txt:149]:
```suggestion
# Check that all XML snippets are on the XML files and not in the RST files so they are tested
```

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: README.md
Comment (EduPonz) [reply README.md:128]: I'm also inclined to that option. However, I'd wait until this PR is merged to do so, cause right now it does not make much sense. After this is merge, we can patch both repos

File: README.md
Comment (EduPonz) [reply README.md:200]: I'm not able to reproduce this

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/837/)

File: CMakeLists.txt
Comment (JLBuenoLopez-eProsima) [reply CMakeLists.txt]: Finally, only `html` builder is going to be supported.

File: CMakeLists.txt
Comment (JLBuenoLopez-eProsima) [reply CMakeLists.txt]: Only `html` builder is going to be supported.

File: Makefile
Comment (JLBuenoLopez-eProsima) [reply Makefile:261]: Only `html` builder is going to be supported.

File: README.md
Comment (JLBuenoLopez-eProsima) [reply README.md]: Only `html` builder is going to be supported.

File: README.md
Comment (JLBuenoLopez-eProsima) [reply README.md:41]: I have not been able to build on Windows. On one hand, CMake is not able to find OpenSSL which is required. I have followed this [instructions](https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_windows.html#openssl) but they are not working for me...

After disabling OpenSSL in the CMakeLists files, the following error appears:

```
CUSTOMBUILD : Configuration error : 
   There is a programmable error in your configuration file:
   OSError: [WinError 126] The specified module could not be found
```

Consequently, I agree to keep the statement that this has only been developed and tested on Ubuntu.

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
Just a couple of small things.

**Review by JLBuenoLopez-eProsima — DISMISSED** [review]
LGTM!

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/840/)

**Comment by richiprosima** [comment]
Buildstatus:[BuildStatus](http://jenkins.eprosima.com:8080/job/fastdds_docs_manual/3/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/841/)

**Comment by richiprosima** [comment]
Buildstatus:[BuildStatus](http://jenkins.eprosima.com:8080/job/fastdds_docs_manual/5/)

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/842/)

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
Good catch!

**Comment by richiprosima** [comment]
Buildstatus:[BuildStatus](http://jenkins.eprosima.com:8080/job/fastdds_docs_manual/6/)

**Comment by richiprosima** [comment]
Buildstatus:[BuildStatus](http://jenkins.eprosima.com:8080/job/fastdds_docs_manual/7/)

**Pull request merged**
